<!doctype html>



  


<html class="theme-next mist use-motion">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  <link href="/vendors/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css"/>




  <link href="//fonts.googleapis.com/css?family=Lato:300,400,700,400italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">



<link href="/vendors/font-awesome/css/font-awesome.min.css?v=4.4.0" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=0.5.0" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="software,workflow,snp,indel," />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=0.5.0" />






<meta name="description" content="GATK(Genome Analysis Toolkit) 是Broad Institute开发高通量测序数据分析的软件包，原先是为了处理千人基因组的数据才开发的，解决了千人基因组测的很多低覆盖度的样本call出来的snp的假阳性太高问题，同时提出了二代测序技术数据处理的一个框架。主要用于突变检测，基因分型，并注重数据质量准确性（低覆盖度数据的校正）">
<meta property="og:type" content="article">
<meta property="og:title" content="GATK工作流程">
<meta property="og:url" content="http://www.bioinfo.site/2016/03/23/gatk-workflow/index.html">
<meta property="og:site_name" content="Bioinfo Notes">
<meta property="og:description" content="GATK(Genome Analysis Toolkit) 是Broad Institute开发高通量测序数据分析的软件包，原先是为了处理千人基因组的数据才开发的，解决了千人基因组测的很多低覆盖度的样本call出来的snp的假阳性太高问题，同时提出了二代测序技术数据处理的一个框架。主要用于突变检测，基因分型，并注重数据质量准确性（低覆盖度数据的校正）">
<meta property="og:image" content="http://www.bioinfo.site/images/gatk_workflow2.png">
<meta property="og:image" content="http://www.bioinfo.site/images/gatk_workflow1.png">
<meta property="og:updated_time" content="2016-03-31T07:07:44.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="GATK工作流程">
<meta name="twitter:description" content="GATK(Genome Analysis Toolkit) 是Broad Institute开发高通量测序数据分析的软件包，原先是为了处理千人基因组的数据才开发的，解决了千人基因组测的很多低覆盖度的样本call出来的snp的假阳性太高问题，同时提出了二代测序技术数据处理的一个框架。主要用于突变检测，基因分型，并注重数据质量准确性（低覆盖度数据的校正）">
<meta name="twitter:image" content="http://www.bioinfo.site/images/gatk_workflow2.png">



<script type="text/javascript" id="hexo.configuration">
  var NexT = window.NexT || {};
  var CONFIG = {
    scheme: 'Mist',
    sidebar: {"position":"right","display":"post"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: 0,
      author: '博主'
    }
  };
</script>

  <title> GATK工作流程 | Bioinfo Notes </title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  



  <script type="text/javascript">
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "//hm.baidu.com/hm.js?87ea9aae27766234564d74b97ebafd19";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>








  
  
    
  

  <div class="container one-collumn sidebar-position-right page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-meta ">
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">Bioinfo Notes</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
  <p class="site-subtitle"></p>
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-home fa-fw"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories" rel="section">
            
              <i class="menu-item-icon fa fa-th fa-fw"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about" rel="section">
            
              <i class="menu-item-icon fa fa-user fa-fw"></i> <br />
            
            关于
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives" rel="section">
            
              <i class="menu-item-icon fa fa-archive fa-fw"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags" rel="section">
            
              <i class="menu-item-icon fa fa-tags fa-fw"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-commonweal">
          <a href="http://101.200.82.139/" rel="section">
            
              <i class="menu-item-icon fa fa-heartbeat fa-fw"></i> <br />
            
            友情链接
          </a>
        </li>
      

      
    </ul>
  

  
</nav>

 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                GATK工作流程
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-03-23T11:07:51+08:00" content="2016-03-23">
              2016-03-23
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/软件/" itemprop="url" rel="index">
                    <span itemprop="name">软件</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
              <span class="post-comments-count">
                &nbsp; | &nbsp;
                <a href="/2016/03/23/gatk-workflow/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count ds-thread-count" data-thread-key="2016/03/23/gatk-workflow/" itemprop="commentsCount"></span>
                </a>
              </span>
            
          

          

          
          

        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        <h4 id="整体流程"><a href="#整体流程" class="headerlink" title="整体流程"></a>整体流程</h4><p>先盗两整体流程框架图，一个是最新的，一个是以前的。整个流程分三个阶段，分别是预处理（pre-processing）、variant discovery和cassset refinement</p>
<p><img src="/images/gatk_workflow2.png" alt="gatk_best_practices_1"></p>
<p><img src="/images/gatk_workflow1.png" alt="gatk_best_practices_1"></p>
<h4 id="预处理（pre-processing）"><a href="#预处理（pre-processing）" class="headerlink" title="预处理（pre-processing）"></a>预处理（pre-processing）</h4><p>预处理的输入文件为原始测序数据（fastaq或uBAM文件均可），并生成用于分析的BAM文件。<br>操作步骤（处理过程）包括比对到基因组和一些数据清理操作使技术（偏见正确？）并使数据便于（后续）分析</p>
<p>index(bwa) -&gt; aln(bwa) -&gt; sampe(bwa) -&gt; view(samtools) -&gt; sortsam(picard-tools) -&gt; MarkDuplicates(picard-tools) -&gt; GenomeAnalysisTK(GATK本地重排，生成重排位置信息) -&gt; GenomeAnalysisTK(GATK本地重排，重排) -&gt; （校正，对低覆盖度的数据很重要）</p>
<h5 id="bwa处理"><a href="#bwa处理" class="headerlink" title="bwa处理"></a>bwa处理</h5><h6 id="参考基因组构建索引（BWA比对）"><a href="#参考基因组构建索引（BWA比对）" class="headerlink" title="参考基因组构建索引（BWA比对）"></a>参考基因组构建索引（BWA比对）</h6><p>GATK要求输入文件是bam格式的，网站上推荐使用bwa进行mapping，bwa速度上比较有优势，而且输出的结果是sam格式的，用samtools转换成bam格式即可。</p>
<p><code>bwa index -a bwtsw ucsc.hg19.fasta</code> （短序列使用-a is）</p>
<h6 id="生成sam格式的比对文件（BWA比对）"><a href="#生成sam格式的比对文件（BWA比对）" class="headerlink" title="生成sam格式的比对文件（BWA比对）"></a>生成sam格式的比对文件（BWA比对）</h6><p><code>bwa mem ref.fasta 1.fq 2.fq &gt; result.01.sam</code></p>
<p><em>注：此样本的测序深度较高，reads读长大于100bp（具体多少？），所以不用其它两种算法（其它两种是什么），而采用mem算法</em></p>
<p>(可以不做？)###### 对sam文件进行重新排序</p>
<p><code>java -jar picard-tools-1.119/ReorderSam.jar I=result.01.sam O=result.reorder.02.sam R=ref.fasta</code></p>
<p>（可以不做？）###### 将sam文件转换成bam文件</p>
<p><code>samtools view -bS result.reorder.02.sam -o result.reorder.03.bam</code></p>
<h5 id="picard处理"><a href="#picard处理" class="headerlink" title="picard处理"></a>picard处理</h5><h6 id="对bam文件进行sort排序处理"><a href="#对bam文件进行sort排序处理" class="headerlink" title="对bam文件进行sort排序处理"></a>对bam文件进行sort排序处理</h6><p>java -jar picard-tools-1.119/SortSam.jar INPUT=result.reorder.03.bam OUTPUT=result.sort.04.bam SORT_ORDER=coordinate</p>
<p>对bam文件进行加头（head）处理</p>
<p>java -jar picard-tools-1.119/AddOrReplaceReadGroups.jar I=/share/data/bioinfo/linsi/project/2015/result/result.sort.04.bam O=/share/data/bioinfo/linsi/project/2015/result/result.add.05.bam ID=group1 LB=lib1 PL=illumina PU=barcode SM=sample1</p>
<ol>
<li>Duplicates Marking （标记过量扩增序列）：</li>
</ol>
<p>java -jar /share/data/bioinfo/linsi/biosoft/picard-tools-1.119/MarkDuplicates.jar  REMOVE_DUPLICATES=false MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 INPUT=/share/data/bioinfo/linsi/project/2015/result/result.add.05.bam OUTPUT=/share/data/bioinfo/linsi/project/2015/result/result.dedup.06.bam METRICS_FILE=/share/data/bioinfo/linsi/project/2015/result/result.dedup.06.metrics</p>
<ol>
<li>生成索引文件：</li>
</ol>
<p>samtools index /share/data/bioinfo/linsi/project/2015/result/result.dedup.06.bam</p>
<ol>
<li><p>Local realignment around indels （将比对到indel附近的reads进行局部重新比对，将比对的错误率降到最低）</p>
<p> 注：以下两命令当源文件为illumina旧版的fastq格式需加参数–fix_misencoded_quality_scores，新版的fastq格式不加</p>
<p> 1) 通过运行RealignerTargetCreator来确定要进行重新比对的区域：</p>
<p> java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -T RealignerTargetCreator –fix_misencoded_quality_scores -I /share/data/bioinfo/linsi/project/2015/result/result.dedup.06.bam -o /share/data/bioinfo/linsi/project/2015/result/result.dedup.07.intervals -known /share/data/bioinfo/linsi/database/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -known /share/data/bioinfo/linsi/database/hg19/1000G_phase1.indels.hg19.sites.vcf</p>
<p> 2) 通过运行IndelRealigner在这些区域内进行重新比对：</p>
<p> java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -T IndelRealigner –fix_misencoded_quality_scores -targetIntervals /share/data/bioinfo/linsi/project/2015/result/result.dedup.07.intervals -I /share/data/bioinfo/linsi/project/2015/result/result.dedup.06.bam -o /share/data/bioinfo/linsi/project/2015/result/result.realn.08.bam -known /share/data/bioinfo/linsi/database/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -known /share/data/bioinfo/linsi/database/hg19/1000G_phase1.indels.hg19.sites.vcf</p>
</li>
<li><p>Base quality score recalibration (BQSR) 对bam文件里reads的碱基质量值进行重新校正，使最后输出的bam文件中reads中碱基的质量值能够更加接近真实的与参考基因组之间错配的概率：</p>
<p> 1) 利用工具BaseRecalibrator，根据一些known sites，生成一个校正质量值所需要的数据文件：</p>
<p> java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T BaseRecalibrator -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -I /share/data/bioinfo/linsi/project/2015/result/result.realn.08.bam -o /share/data/bioinfo/linsi/project/2015/result/result.realn.09_1.grp -knownSites /share/data/bioinfo/linsi/database/hg19/dbsnp_138.hg19.vcf -knownSites /share/data/bioinfo/linsi/database/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -knownSites /share/data/bioinfo/linsi/database/hg19/1000G_phase1.indels.hg19.sites.vcf</p>
<p> 2) 利用工具PrintReads将经过质量值校正的数据输出到新的bam文件中，用于后续的变异检测：</p>
<p> java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T PrintReads -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -I /share/data/bioinfo/linsi/project/2015/result/result.realn.08.bam -BQSR /share/data/bioinfo/linsi/project/2015/result/result.realn.09_1.grp -o /share/data/bioinfo/linsi/project/2015/result/result.realn.09_3.grp.bam</p>
</li>
<li><p>calling variants with haplotypecaller (生成vcf文件)</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T HaplotypeCaller -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -I /share/data/bioinfo/linsi/project/2015/result/result.realn.09_3.grp.bam –genotyping_mode DISCOVERY –output_mode EMIT_VARIANTS_ONLY -stand_emit_conf 10 -stand_call_conf 30 -o /share/data/bioinfo/linsi/project/2015/result/result.10.raw_variants.vcf</p>
</li>
<li><p>variant quality score recalibration (VQSR) </p>
<p>1) 建立SNP校正模型（这一步可以尽可能多的设置注释类型和tranche的值，然后根据这次跑出来的结果选择出最好的注释类型和tranche值之后，再次运行VariantRecalibrator）：</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T VariantRecalibrator -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta –maxGaussians 4 -input /share/data/bioinfo/linsi/project/2015/result/result.10.raw_variants.vcf -resource:hapmap,known=false,training=true,truth=true,prior=15.0 /share/data/bioinfo/linsi/database/hg19/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 /share/data/bioinfo/linsi/database/hg19/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 /share/data/bioinfo/linsi/database/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 /share/data/bioinfo/linsi/database/hg19/dbsnp_138.hg19.vcf -an DP -an MQ -an QD -mode SNP -tranche 99.9 -tranche 99.0 -tranche 97.0 -tranche 95.0 -tranche 93.0 -minNumBad 1000 -recalFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.recal -tranchesFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.tranches -rscriptFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.R</p>
<p>2) 这一步run出来的结果可以直接用于下一步的ApplyRecalibration：</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T VariantRecalibrator -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta –maxGaussians 4 -input /share/data/bioinfo/linsi/project/2015/result/result.10.raw_variants.vcf -mode SNP -resource:hapmap,known=false,training=true,truth=true,prior=15.0 /share/data/bioinfo/linsi/database/hg19/hapmap_3.3.hg19.sites.vcf -resource:omni,known=false,training=true,truth=false,prior=12.0 /share/data/bioinfo/linsi/database/hg19/1000G_omni2.5.hg19.sites.vcf -resource:1000G,known=false,training=true,truth=false,prior=10.0 /share/data/bioinfo/linsi/database/hg19/1000G_phase1.snps.high_confidence.hg19.sites.vcf -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 /share/data/bioinfo/linsi/database/hg19/dbsnp_138.hg19.vcf -an DP -an MQ -an QD –TStranche 99.0 -recalFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.real.recal -tranchesFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.real.tranches -rscriptFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.real.R</p>
<p>3) 根据上步得到的校正水平做SNP校正，此时文件中的SNP位点已经全部经过校正过滤：</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -T ApplyRecalibration -mode SNP -input /share/data/bioinfo/linsi/project/2015/result/result.10.raw_variants.vcf -recalFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.real.recal -tranchesFile /share/data/bioinfo/linsi/project/2015/result/result.10.vcf.real.tranches -o /share/data/bioinfo/linsi/project/2015/result/result.10.recalibrated_snps_raw_indels.vcf</p>
<p>4) 对INDEL结果进行校正，同SNP多选择一些注释类型，但是不用选择tranche值,tranche值是专门为SNP设定的</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -T VariantRecalibrator -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -mode INDEL –maxGaussians 4 -std 10.0 -input /share/data/bioinfo/linsi/project/2015/result/result.10.recalibrated_snps_raw_indels.vcf -resource:mills,known=true,training=true,truth=true,prior=12.0 /share/data/bioinfo/linsi/database/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -an DP -an MQ -an QD -an FS -an MQRankSum -recalFile /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrate.indel.recal -tranchesFile /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrate.indel.tranches -rscriptFile /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrate.indel.R</p>
<p>5) 根据上步得到的校正水平做INDEL校正</p>
<p>java -jar /share/biosoft/apps/GTK3.4-46/GenomeAnalysisTK.jar -R /share/data/bioinfo/linsi/database/hg19/ucsc.hg19.fasta -T ApplyRecalibration -mode INDEL -input /share/data/bioinfo/linsi/project/2015/result/result.10.recalibrated_snps_raw_indels.vcf -recalFile /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrate.indel.recal -tranchesFile /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrate.indel.tranches -o /share/data/bioinfo/linsi/project/2015/result/result.11.recalibrated_variants.vcf</p>
</li>
</ol>
<h5 id="variant-discovery"><a href="#variant-discovery" class="headerlink" title="variant discovery"></a>variant discovery</h5><p>变异检测的输入文件为待分析的BAM文件，并生成VCF格式的callset。操作过程包括识别可能的基因突变位点，并运用过滤方法以符合实验设计要求</p>
<h5 id="callset-refinement"><a href="#callset-refinement" class="headerlink" title="callset refinement"></a>callset refinement</h5><p>callset refniement 的输入文件为vcf格式的callset，操作过程包括使用元数据估计和提高基因分型正确性，添加额外信息并评价callset的整体质量</p>
<h6 id="参考资料"><a href="#参考资料" class="headerlink" title="参考资料"></a>参考资料</h6><p><a href="https://www.broadinstitute.org/gatk/guide/best-practices.php" target="_blank" rel="external">GATK Best Practices</a><br><a href="http://www.plob.org/2014/02/21/7070.html" target="_blank" rel="external">GATK使用</a><br><a href="http://blog.163.com/zhoulili1987619@126/blog/static/3530820120139158521986/" target="_blank" rel="external">GATK使用简介</a></p>

      
    </div>

    <div>
      
        
      
    </div>

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/software/" rel="tag">#software</a>
          
            <a href="/tags/workflow/" rel="tag">#workflow</a>
          
            <a href="/tags/snp/" rel="tag">#snp</a>
          
            <a href="/tags/indel/" rel="tag">#indel</a>
          
        </div>
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2016/03/23/linux-tips/" rel="next" title="linux 小技巧">
                <i class="fa fa-chevron-left"></i> linux 小技巧
              </a>
            
          </div>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2016/03/25/perl-tips1/" rel="prev" title="perl 备记录--生信1（持续更新）">
                perl 备记录--生信1（持续更新） <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div class="ds-thread" data-thread-key="2016/03/23/gatk-workflow/"
           data-title="GATK工作流程" data-url="http://www.bioinfo.site/2016/03/23/gatk-workflow/">
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel ">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/images/ted_suit.png"
               alt="Samuel Lin" />
          <p class="site-author-name" itemprop="name">Samuel Lin</p>
          <p class="site-description motion-element" itemprop="description">we drank and talked for an hour</p>
        </div>
        <nav class="site-state motion-element">
          <div class="site-state-item site-state-posts">
            <a href="/archives">
              <span class="site-state-item-count">9</span>
              <span class="site-state-item-name">日志</span>
            </a>
          </div>

          
            <div class="site-state-item site-state-categories">
              <a href="/categories">
                <span class="site-state-item-count">3</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            <div class="site-state-item site-state-tags">
              <a href="/tags">
                <span class="site-state-item-count">19</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
        </div>

        
        

        
        <div class="links-of-blogroll motion-element">
          
        </div>

      </section>

      
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc-indicator-top post-toc-indicator">
            <i class="fa fa-angle-double-up"></i>
          </div>
          <div class="post-toc">
            
              
            
            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-4"><a class="nav-link" href="#整体流程"><span class="nav-number">1.</span> <span class="nav-text">整体流程</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#预处理（pre-processing）"><span class="nav-number">2.</span> <span class="nav-text">预处理（pre-processing）</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#bwa处理"><span class="nav-number">2.1.</span> <span class="nav-text">bwa处理</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#参考基因组构建索引（BWA比对）"><span class="nav-number">2.1.1.</span> <span class="nav-text">参考基因组构建索引（BWA比对）</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#生成sam格式的比对文件（BWA比对）"><span class="nav-number">2.1.2.</span> <span class="nav-text">生成sam格式的比对文件（BWA比对）</span></a></li></ol></li><li class="nav-item nav-level-5"><a class="nav-link" href="#picard处理"><span class="nav-number">2.2.</span> <span class="nav-text">picard处理</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#对bam文件进行sort排序处理"><span class="nav-number">2.2.1.</span> <span class="nav-text">对bam文件进行sort排序处理</span></a></li></ol></li><li class="nav-item nav-level-5"><a class="nav-link" href="#variant-discovery"><span class="nav-number">2.3.</span> <span class="nav-text">variant discovery</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#callset-refinement"><span class="nav-number">2.4.</span> <span class="nav-text">callset refinement</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#参考资料"><span class="nav-number">2.4.1.</span> <span class="nav-text">参考资料</span></a></li></ol></li></ol></li></ol></div>
            
          </div>
          <div class="post-toc-indicator-bottom post-toc-indicator">
            <i class="fa fa-angle-double-down"></i>
          </div>
        </section>
      

    </div>
  </aside>


        
      </div>
    </main>
<!--
    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2016</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Samuel Lin</span>
</div>

<div class="powered-by">
  由 <a class="theme-link" href="http://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


      </div>
    </footer>
-->
    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  


  




<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>


  <script type="text/javascript" src="/vendors/jquery/index.js?v=2.1.3"></script>

  <script type="text/javascript" src="/vendors/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  <script type="text/javascript" src="/vendors/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  <script type="text/javascript" src="/vendors/velocity/velocity.min.js"></script>

  <script type="text/javascript" src="/vendors/velocity/velocity.ui.min.js"></script>

  <script type="text/javascript" src="/vendors/fancybox/source/jquery.fancybox.pack.js"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=0.5.0"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=0.5.0"></script>



  
  

  
  
<script type="text/javascript" src="/js/src/scrollspy.js?v=0.5.0"></script>

<script type="text/javascript" id="sidebar.toc.highlight">
  $(document).ready(function () {
    var tocSelector = '.post-toc';
    var $tocSelector = $(tocSelector);
    var activeCurrentSelector = '.active-current';

    $tocSelector
      .on('activate.bs.scrollspy', function () {
        var $currentActiveElement = $(tocSelector + ' .active').last();

        removeCurrentActiveClass();
        $currentActiveElement.addClass('active-current');

        $tocSelector[0].scrollTop = $currentActiveElement.position().top;
      })
      .on('clear.bs.scrollspy', function () {
        removeCurrentActiveClass();
      });

    function removeCurrentActiveClass () {
      $(tocSelector + ' ' + activeCurrentSelector)
        .removeClass(activeCurrentSelector.substring(1));
    }

    function processTOC () {
      getTOCMaxHeight();
      toggleTOCOverflowIndicators();
    }

    function getTOCMaxHeight () {
      var height = $('.sidebar').height() -
                   $tocSelector.position().top -
                   $('.post-toc-indicator-bottom').height();

      $tocSelector.css('height', height);

      return height;
    }

    function toggleTOCOverflowIndicators () {
      tocOverflowIndicator(
        '.post-toc-indicator-top',
        $tocSelector.scrollTop() > 0 ? 'show' : 'hide'
      );

      tocOverflowIndicator(
        '.post-toc-indicator-bottom',
        $tocSelector.scrollTop() >= $tocSelector.find('ol').height() - $tocSelector.height() ? 'hide' : 'show'
      )
    }

    $(document).on('sidebar.motion.complete', function () {
      processTOC();
    });

    $('body').scrollspy({ target: tocSelector });
    $(window).on('resize', function () {
      if ( $('.sidebar').hasClass('sidebar-active') ) {
        processTOC();
      }
    });

    onScroll($tocSelector);

    function onScroll (element) {
      element.on('mousewheel DOMMouseScroll', function (event) {
          var oe = event.originalEvent;
          var delta = oe.wheelDelta || -oe.detail;

          this.scrollTop += ( delta < 0 ? 1 : -1 ) * 30;
          event.preventDefault();

          toggleTOCOverflowIndicators();
      });
    }

    function tocOverflowIndicator (indicator, action) {
      var $indicator = $(indicator);
      var opacity = action === 'show' ? 1 : 0;
      $indicator.velocity ?
        $indicator.velocity('stop').velocity({
          opacity: opacity
        }, { duration: 100 }) :
        $indicator.stop().animate({
          opacity: opacity
        }, 100);
    }

  });
</script>

<script type="text/javascript" id="sidebar.nav">
  $(document).ready(function () {
    var html = $('html');
    var TAB_ANIMATE_DURATION = 200;
    var hasVelocity = $.isFunction(html.velocity);

    $('.sidebar-nav li').on('click', function () {
      var item = $(this);
      var activeTabClassName = 'sidebar-nav-active';
      var activePanelClassName = 'sidebar-panel-active';
      if (item.hasClass(activeTabClassName)) {
        return;
      }

      var currentTarget = $('.' + activePanelClassName);
      var target = $('.' + item.data('target'));

      hasVelocity ?
        currentTarget.velocity('transition.slideUpOut', TAB_ANIMATE_DURATION, function () {
          target
            .velocity('stop')
            .velocity('transition.slideDownIn', TAB_ANIMATE_DURATION)
            .addClass(activePanelClassName);
        }) :
        currentTarget.animate({ opacity: 0 }, TAB_ANIMATE_DURATION, function () {
          currentTarget.hide();
          target
            .stop()
            .css({'opacity': 0, 'display': 'block'})
            .animate({ opacity: 1 }, TAB_ANIMATE_DURATION, function () {
              currentTarget.removeClass(activePanelClassName);
              target.addClass(activePanelClassName);
            });
        });

      item.siblings().removeClass(activeTabClassName);
      item.addClass(activeTabClassName);
    });

    $('.post-toc a').on('click', function (e) {
      e.preventDefault();
      var targetSelector = NexT.utils.escapeSelector(this.getAttribute('href'));
      var offset = $(targetSelector).offset().top;
      hasVelocity ?
        html.velocity('stop').velocity('scroll', {
          offset: offset  + 'px',
          mobileHA: false
        }) :
        $('html, body').stop().animate({
          scrollTop: offset
        }, 500);
    });

    // Expand sidebar on post detail page by default, when post has a toc.
    NexT.motion.middleWares.sidebar = function () {
      var $tocContent = $('.post-toc-content');

      if (CONFIG.sidebar.display === 'post' || CONFIG.sidebar.display === 'always') {
        if ($tocContent.length > 0 && $tocContent.html().trim().length > 0) {
          NexT.utils.displaySidebar();
        }
      }
    };
  });
</script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=0.5.0"></script>



  

  
    
  

  <script type="text/javascript">
    var duoshuoQuery = {short_name:"samuellin"};
    (function() {
      var ds = document.createElement('script');
      ds.type = 'text/javascript';ds.async = true;
      ds.id = 'duoshuo-script';
      ds.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') + '//static.duoshuo.com/embed.js';
      ds.charset = 'UTF-8';
      (document.getElementsByTagName('head')[0]
      || document.getElementsByTagName('body')[0]).appendChild(ds);
    })();
  </script>

  
    
  





  
  
  

  


</body>
</html>
